src/Forms/ConfirmedPasswordField.php

```diff
@@ -155,6 +177,47 @@ public function __construct(
         $this->setValue($value);
     }
 
+    /**
+     * Endpoint to provide feedback for the current and required level of password strength
+     */
+    public function strength(HTTPRequest $request): HTTPResponse
+    {
+        $response = HTTPResponse::create();
+        $json = json_decode($request->getBody() ?? '', true);
+        if (!$json || !array_key_exists('password', $json) || !$request->isPOST()) {
+            $response->setStatusCode(400);
+            return $response;
+        }
+        $password = $json['password'];
+        $strength = -1;
+        $strengthLabel = '';
+        $validator = $this->getApplicableValidator();
+        if (is_a($validator, EntropyPasswordValidator::class)) {
+            $strength = PasswordStrengthValidator::estimateStrength($password);
```

Comment:
```suggestion
        if ($this->getMinPasswordStrengthForEvaluation() >= 0) {
            $strength = PasswordStrengthValidator::estimateStrength($password);
```
We don't actually need a member `PasswordValidator` at all. All of the logic for this condition is already in `getMinPasswordStrengthForEvaluation()` - i.e. one of:

1. `requireStrongPassword()` is true
1. we're on a member form and the current member validator is a `EntropyPasswordValidator`

Reply:
Testing with getMinPasswordStrengthForEvaluation() looks really weird given that this is an endpoint

Also it'll mess up IDE type-hinting because the `if (is_a($validator, EntropyPasswordValidator::class)) {` makes the IDE know that `$validator->getStrengthLabel($strength);` is OK to call

Reply:
The current implementation is wrong. It will stop users from getting an esimation of their password strength for a valid `ConfirmedPasswordField::create()->setRequiresMinPasswordStrength(true)` if the _member_'s `PasswordValidator` is not an entropy password validator.

Reply:
Updated